src/wasi.rs

```diff
@@ -0,0 +1,792 @@
+pub use ffi::c_void;
+
+pub type c_char = i8;
+pub type c_uchar = u8;
+pub type c_int = i32;
+pub type c_uint = u32;
+pub type c_short = i16;
+pub type c_ushort = u16;
+pub type c_long = i32;
+pub type c_ulong = u32;
+pub type c_longlong = i64;
+pub type c_ulonglong = u64;
+pub type size_t = usize;
+pub type ssize_t = isize;
+pub type off_t = i64;
+pub type pid_t = i32;
+pub type int8_t = i8;
```

Comment:
I'd rather not add the `int8_t` and similar type-aliases. We do guarantee that `i8 == int8_t` on all targets, and people should just use `i8` on C FFI in the same way they just use `bool` - we should just deprecate these aliases instead for the other targets (see #1304 ).

Reply:
This has come up historically though where for better or worse these types are used in standard Rust types and whatnot, so I think it's better to include them on all platforms for better compatibility and consider the deprecation story separately?

Reply:
We definitely can do that. I think that some projects use these and some projects don't. Changing a project from using `int8_t` to just using `i8` is a "backwards compatible" change, and if someone complaints that wasi does not have these we can always add them later.

Reply:
And by that I mean that I'll leave it up to you.

Reply:
Ok in that case I think I'd be inclined to go ahead and add them because we've already had at least one expicit request to add it to the wasm32-unknown-unknown target (which this still doesn't do)

Reply:
Sure, fine with me. I'm marking this as resolved. 

Do you have a link to the request ? I think the biggest thing blocking the deprecation is telling bindgen to just use the Rust types always.

Reply:
Ah not currently, I forget where the request came from...